tests/sentry/api/path_params/resources/test_project_slug.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any
+from unittest.mock import MagicMock
+
+from rest_framework.request import Request
+
+from sentry.testutils.cases import TestCase
+
+from .test_id_or_slug_path_params_mixin import APIIdOrSlugTestMixin
+
+
+class ProjectSlugTests(TestCase, APIIdOrSlugTestMixin):
+    def project_test(self, endpoint_class, slug_params, *args):
+        slug_kwargs = {param: self.slug_mappings[param].slug for param in slug_params}
+        id_kwargs = {param: self.slug_mappings[param].id for param in slug_params}
+
+        request = Request(self.make_request(user=self.user))
+
+        _, converted_slugs = endpoint_class().convert_args(request=request, **slug_kwargs)
+        _, converted_ids = endpoint_class().convert_args(request=request, **id_kwargs)
+
+        self.assert_conversion(endpoint_class, converted_slugs, converted_ids)
+
+    def project_alert_rule_endpoint_test(self, endpoint_class, slug_params, *args):
+        slug_kwargs = {param: self.slug_mappings[param].slug for param in slug_params}
+        id_kwargs = {param: self.slug_mappings[param].id for param in slug_params}
+
+        request = MagicMock()
+        request.configure_mock(
+            **{
+                "access.has_project_access.return_value": True,
+                "method": "DELETE",
+            }
+        )
+
+        alert_rule = self.create_alert_rule(organization=self.organization, projects=[self.project])
+
+        non_slug_mappings: dict[str, Any] = {
+            "alert_rule_id": alert_rule.id,
+        }
+
+        reverse_non_slug_mappings: dict[str, Any] = {
```

Comment:
extreme nit: This is very nit, but can we group `non_slug_mappings` and `reverse_non_slug_mappings` together without the whitespace like you're doing for `converted_ids/slugs`. There's just a lot of tests on the page and it sorta helps visually 😶
```suggestion
        }
        reverse_non_slug_mappings: dict[str, Any] = {
```